v3/src/components/app.tsx

```diff
@@ -65,22 +75,26 @@ export const App = observer(function App() {
       })
       // return to "normal" after import process is complete
       sharedData?.dataSet.completeSnapshot()
-    }, [])
+      onCloseUserEntry()
+    }, [onCloseUserEntry])
 
   const handleImportDocument = useCallback((file: File) => {
     cfm?.client.openLocalFileWithConfirmation(file)
-  }, [cfm])
+    onCloseUserEntry()
+  }, [cfm, onCloseUserEntry])
 
   const handleUrlDrop = useCallback((url: string) => {
     const tile = appState.document.content?.createOrShowTile(kWebViewTileType)
     isWebViewModel(tile?.content) && tile?.content.setUrl(url)
-  }, [])
+    onCloseUserEntry()
+  }, [onCloseUserEntry])
 
   useDropHandler({
-    selector: `#${kCodapAppElementId}`,
+    selector: isOpenUserEntry ? `#${kUserEntryDropOverlay}` : `#${kCodapAppElementId}`,
     onImportDataSet: handleImportDataSet,
     onImportDocument: handleImportDocument,
-    onHandleUrlDrop: handleUrlDrop
+    onHandleUrlDrop: handleUrlDrop,
+    setIsDragOver
```

Comment:
```suggestion
    onSetIsDragOver: setIsDragOver
```
Conventional naming for callbacks.